frontend/src/hooks/useCreateBatch.ts

```diff
@@ -0,0 +1,82 @@
+import { useSelectionStore, useProjectStore } from '/src/stores'
+
+import { CopyData, Template } from '/src/types/ProjectTypes'
+import { PASTE_POSITION_OFFSET } from '/src/config/rendering'
+
+const useCreateBatch = (createData: CopyData | Template) => {
+  const project = useProjectStore(s => s.project)
+  const createState = useProjectStore(s => s.createState)
+  const createComment = useProjectStore(s => s.createComment)
+  const createTransition = useProjectStore(s => s.createTransition)
+  const removeStates = useProjectStore(s => s.removeStates)
+  const selectStates = useSelectionStore(s => s.setStates)
+  const selectTransitions = useSelectionStore(s => s.setTransitions)
+  const selectComments = useSelectionStore(s => s.setComments)
+  const setStateInitial = useProjectStore(s => s.setStateInitial)
+  const commit = useProjectStore(s => s.commit)
+
+  const createBatch = () => {
```

Comment:
I tried to make a hook for inserting + selecting a group of elements (states, transitions, comments) as this is used in the copy and template insert functions. I couldn't get it to work properly because I always had issues calling so I made it into a store method insertGroup in the useProjectStore. I would need some help/pairing to make this happen, but for now it works in the store although it doesn't seem like the best solution

Reply:
(to clarify, this isn't actually called anywhere but I have left it in for now in case)

Reply:
If you don't need it anymore then feel free to delete 👍
We can recover it through git anyways if we need it later